include/seahorn/Bmc.hh

```diff
@@ -14,20 +21,66 @@
 
 namespace seahorn {
 using namespace expr;
+using ZModel_ref = std::shared_ptr<ZModel<EZ3>>;
 
 namespace bmc_impl {
 /// true if I is a call to a void function
 bool isCallToVoidFn(const llvm::Instruction &I);
 /// computes an implicant of f (interpreted as a conjunction) that
 /// contains the given model
-void get_model_implicant(const ExprVector &f, ZModel<EZ3> &model,
-                         ExprVector &out, ExprMap &active_bool_map);
+template <typename model_ref>
```

Comment:
minimize code that lives in the header file used by clients. 
`Bmc.hh` is included by clients of Bmc engines. Clients should not import implementation. 

Put implementation into another header file that is only included by `Bmc.cc`. Usual name is `BmcImpl.hh` or some similar convention